flink-core/src/main/java/org/apache/flink/configuration/CheckpointingOptions.java

```diff
@@ -157,4 +157,14 @@
 		.withDescription(String.format("The default size of the write buffer for the checkpoint streams that write to file systems. " +
 			"The actual write buffer size is determined to be the maximum of the value of this option and option '%s'.", FS_SMALL_FILE_THRESHOLD.key()));
 
+	/**
+	 * The default size of the read buffer for the checkpoint streams that read from file systems.
+	 */
+	@Documentation.Section(Documentation.Sections.EXPERT_STATE_BACKENDS)
+	public static final ConfigOption<Integer> FS_READ_BUFFER_SIZE = ConfigOptions
+		.key("state.backend.fs.read-buffer-size")
+		.intType()
```

Comment:
Shouldn't the type of this be `memory`? I know some of the others aren't but that's because they predate us having a memory configuration type.

Reply:
This configuration only means reading a buffer size of the file system. Should the `memory configuration` contain some `memory size` related config?

Reply:
I am also in favor of `memory type` as the buffer is just an array of bytes allocated from JVM memory. The reason why `state.backend.fs.write-buffer-size` is integer type is that it implemented before Flink introduced `memory type` configuration.

Reply:
done